src/components/Lobby/index.js

```diff
@@ -52,6 +54,18 @@ export default function Lobby() {
     [chatMessage, displayName, socket],
   );
 
+  const handleRoomClick = useCallback(
+    (roomId) => {
+      const targetUser = connectedUsers[roomId];
+      if (targetUser && Object.keys(targetUser).length === 2) {
+        alert("Room is full");
+      } else {
+        navigate(`/battles/${roomId}`);
```

Comment:
```suggestion
      const selectedRoom = connectedUsers[roomId];
      if (selectedRoom && Object.keys(selectedRoom).length === 2) {
        return alert("Room is full");
      } 
       
       return navigate(`/battles/${roomId}`);
```
그리고 추가적으로 받아온 `connectedUsers` 객체를 `redux`에서 관리하는 것은 어떨까요?